sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -4007,25 +4127,107 @@ def hpat_pandas_series_lt(self, other, level=None, fill_value=None, axis=0):
         ty_checker.raise_exc(level, 'None', 'level')
 
-    if not (isinstance(fill_value, types.Omitted) or fill_value is None):
-        ty_checker.raise_exc(fill_value, 'None', 'fill_value')
+    if not isinstance(fill_value, (types.Omitted, types.Number, types.NoneType)) and fill_value is not None:
+        ty_checker.raise_exc(fill_value, 'number', 'fill_value')
 
     if not (isinstance(axis, types.Omitted) or axis == 0):
         ty_checker.raise_exc(axis, 'int', 'axis')
 
-    if isinstance(other, SeriesType):
-        def hpat_pandas_series_lt_impl(self, other, level=None, fill_value=None, axis=0):
-            return pandas.Series(self._data < other._data)
+    self_is_series, other_is_series = isinstance(self, SeriesType), isinstance(other, SeriesType)
+    if not (self_is_series or other_is_series):
+        return None
 
-        return hpat_pandas_series_lt_impl
+    if not isinstance(self, (SeriesType, types.Number, types.UnicodeType)):
+        ty_checker.raise_exc(self, 'pandas.series or scalar', 'self')
 
-    if isinstance(other, types.Number):
-        def hpat_pandas_series_lt_impl(self, other, level=None, fill_value=None, axis=0):
-            return pandas.Series(self._data < other)
+    if not isinstance(other, (SeriesType, types.Number, types.UnicodeType)):
+        ty_checker.raise_exc(other, 'pandas.series or scalar', 'other')
 
-        return hpat_pandas_series_lt_impl
+    operands_are_series = self_is_series and other_is_series
+    if operands_are_series:
+        none_or_numeric_indexes = ((isinstance(self.index, types.NoneType) or check_index_is_numeric(self))
+                                   and (isinstance(other.index, types.NoneType) or check_index_is_numeric(other)))
+        series_indexes_comparable = check_types_comparable(self.index, other.index) or none_or_numeric_indexes
+        if not series_indexes_comparable:
+            raise TypingError('{} Not implemented for series with not-comparable indexes. \
+            Given: self.index={}, other.index={}'.format(_func_name, self.index, other.index))
 
-    ty_checker.raise_exc(other, 'Series, int, float', 'other')
+    series_data_comparable = check_types_comparable(self, other)
+    if not series_data_comparable:
+        raise TypingError('{} Not supported for not-comparable operands. \
+        Given: self={}, other={}'.format(_func_name, self, other))
+
+    fill_value_is_none = False
+    if isinstance(fill_value, (types.NoneType, types.Omitted)) or fill_value is None:
+        fill_value_is_none = True
+    if not operands_are_series:
+        def _series_lt_scalar_impl(self, other, level=None, fill_value=None, axis=0):
+            fill_value_is_nan = False
+            if fill_value is None:
+                fill_value = numpy.nan
+            if not fill_value_is_none == True:  # noqa
+                fill_value_is_nan = numpy.isnan(fill_value)
+            if not (fill_value_is_nan or fill_value_is_none == True):  # noqa
```

Comment:
You do not need to capture fill_value_is_none as compile time constant, as there's no need to eliminate dead branches here. If all you need is to run fillna when fill_value is not None/np.nan, you can just write:
```
            if (fill_value is not None and not numpy.isnan(fill_value)):
                numpy_like.fillna(self._data, inplace=True, value=fill_value)
```